File: frontend/src/types/weather.ts
export interface WeatherData {
  temperature: number;
  condition: string;
  humidity: number;
  wind_speed: number;
  location: string;
  precipitation: number;
  fallback?: boolean;
}

export interface WeatherRequest {
  location: string;
}

export interface WeatherApiResponse extends WeatherData {
  success?: boolean;
  error?: string;
}

export type WeatherCondition = 
  | "Clear"
  | "Sunny" 
  | "Partly Cloudy"
  | "Cloudy"
  | "Overcast"
  | "Rain"
  | "Light Rain"
  | "Heavy Rain"
  | "Thunderstorm"
  | "Snow"
  | "Light Snow"
  | "Heavy Snow"
  | "Fog"
  | "Mist"
  | "Windy"
  | "Hot"
  | "Cold";

export interface UserLocation {
  city?: string;
  coordinates?: {
    latitude: number;
    longitude: number;
  };
  timezone?: string;
}

File: src/custom_types/weather.py
from pydantic import BaseModel
from typing import Optional

class WeatherData(BaseModel):
    temperature: float
    condition: str
    humidity: float
    wind_speed: Optional[float] = None
    location: Optional[str] = None
    precipitation: float = 0.0

# Export types for use in other backend modules
__all__ = ['WeatherData'] 